### PR TITLE
Add a changelog for all packages

### PR DESCRIPTION
**Pull request by Luthaf**
----
:books: Documentation preview :books:: https://metatensor--403.org.readthedocs.build/en/403/

### CODE REVIEW

File: metatensor-core/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+### metatensor-core Python
+
+#### Added
+
+#### Fixed
+
+#### Changed
+
+#### Removed
```
Comment (PicoCentauri) [inline metatensor-core/CHANGELOG.md:38]: Do we really want it to be that detailed? If yes, the rust part missing, right?

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Yes will be useful for the future. I would also add a `PULL_REQUEST_TEMPLATE.md` to the `.github` and do we also want the changelog to be displayed in the docs?

```md
<!-- What does this implement/fix? Explain your changes here. -->


Contributor (creator of PR) checklist
-------------------------------------
 - [ ] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - [ ] Issue referenced (for PRs that solve an issue)?

For Reviewer
------------
 - [ ] CHANGELOG updated if important change?
 ```

File: metatensor-core/CHANGELOG.md (hunk 1 above)
Comment (Luthaf) [reply metatensor-core/CHANGELOG.md:38]: I'd say yes, we want to document all changes in public API. This is also why the rust part is not here. For metatensor-core, using Rust is an implementation detail.

**Comment by Luthaf** [comment]
> Yes will be useful for the future. I would also add a PULL_REQUEST_TEMPLATE.md to the .github and do we also want the changelog to be displayed in the docs?

Yes to both, I'll have a go at it ASAP

File: docs/src/reference/operations/index.rst [hunk 2]
```diff
@@ -52,3 +52,5 @@ List of all operations
     manipulation/index
     math/index
     set/index
+
+    CHANGELOG.md
```
Comment (PicoCentauri) [inline docs/src/reference/operations/index.rst:56]: Why a new line here?

```suggestion
    CHANGELOG.md
```

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Thanks for the chnages! Already looks super good. I have some minor improvements and ideas.

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**